src/NFTMarkerCreator.js

```diff
@@ -318,26 +316,22 @@ async function useJPG_w_sharp(buf) {
             imageData.sizeX = metadata.width;
             imageData.sizeY = metadata.height;
             imageData.nc = metadata.channels;
-            return image.toBuffer()
+            return image
+            .raw()
+            .toBuffer()
         })   
         .then(data => {
-            let newArr = [];
+            let dt = data.buffer
 
-            let verifyColorSpace = detectColorSpace(data);
+            let verifyColorSpace = detectColorSpace(dt);
 
             if (verifyColorSpace === 1) {
-                for (let j = 0; j < data.length; j += 4) {
```

Comment:
i removed these lines because are not necessary in my opinion, we can pass the data buffer directly to the Uint8Array, we simply log a message in the console.